README.md

```diff
@@ -150,7 +150,7 @@ const { flags, values, positionals } = parseArgs(argv, options);
 ### F.A.Qs
 
 - Is `cmd --foo=bar baz` the same as `cmd baz --foo=bar`?
-  - Yes, if `withValue: ['foo']`, otherwise no
+  - yes
```

Comment:
Some commands force positional to come after dashed args. Is this possible to configure?

Reply:
Short version: no, not possible to configure.

For interest, ordering is a difference between the POSIX specification and what GNU does.

https://www.gnu.org/prep/standards/html_node/Command_002dLine-Interfaces.html

> It is a good idea to follow the POSIX guidelines for the command-line options of a program. The easiest way to do this is to use getopt to parse them. Note that the GNU version of getopt will normally permit options anywhere among the arguments unless the special argument ‘--’ is used. This is not what POSIX specifies; it is a GNU extension.

Reply:
`git` forces this; i think people will be surprised if it's not configurable.

Reply:
There is a difference between ordering of options relative to subcommands and ordering of options relative to positional.

I did a quick check and not seeing `git` enforcing order of options and positional:

```
% git switch --no-guess main
Switched to branch 'main'
Your branch is up to date with 'upstream/main'.

% git switch feature/basic-short
Switched to branch 'feature/basic-short'
Your branch is up to date with 'origin/feature/basic-short'.

% git switch main --no-guess
Switched to branch 'main'
Your branch is up to date with 'upstream/main'.
```

Reply:
I've never used `git switch`; try `git diff package.json -w` and you'll get `fatal: option '-w' must come before non-option arguments`.

Reply:
Digging into the git behaviour. `git switch` is fairly new. I do see the error with `git diff`, interesting thanks. I tried another "old" command, and again ordering not enforced:

```
 % git clone git@github.com:tj/commander.js.git --bare    
Cloning into bare repository 'commander.js.git'...
```

I speculate that there is potential for confusion with "path" and options, so `git diff` has different rules. The help does explicitly show `[--]` on each of the variations that may take a "path". 

```
SYNOPSIS
       git diff [<options>] [<commit>] [--] [<path>...]
       git diff [<options>] --cached [--merge-base] [<commit>] [--] [<path>...]
       git diff [<options>] [--merge-base] <commit> [<commit>...] <commit> [--] [<path>...]
       git diff [<options>] <commit>...<commit> [--] [<path>...]
       git diff [<options>] <blob> <blob>
       git diff [<options>] --no-index [--] <path> <path>
```

Reply:
Looking at the Big Four parsing libraries, all of them support options after positional by default. None of them support making it an error.

Three support a mode where parsing stops after the first non-option, so the remaining arguments can be passed through to another command (say).

- minimist: `opts.stopEarly`
- yargs: `halt-at-non-option`
- commander: `.passThroughOptions()`

By default `argParse` allows options before positionals, or options after positionals, but will display an error for inter-mixed options and positionals. You can opt-in to mixing them by using `parser.parse_intermixed_args()`. The opt-in is because intermixing interacts with a few of the other opt-in features.